zbMATH.js

```diff
@@ -35,105 +35,163 @@
+		}
+
+		function pushWord(element) {
+			let word = cleanText(element);
+			if (word) {
+				item.tags.push(word);
 			}
-			
-			// keywords are normally not in the bib file, so we take them from the page
-			// moreover, the meaning of the MSC classification is also only given on the page
-			if (item.tags.length == 0) {
-				var keywords = ZU.xpath(doc, '//div[@class="keywords"]/a');
-				for (var i = 0; i < keywords.length; i++) {
-					item.tags.push(keywords[i].textContent);
+		}
+
+		// Keywords
+		doc.querySelectorAll(".keywords a").forEach(pushWord);
+		// Labels to MSC identifiers. Don't put the replaced MathJax/MathML
+		// in $ $, because this isn't very useful in tags.
+		doc.querySelectorAll(".classification td.space")
+			.forEach(node => pushWord(cleanupMath(node, false/* laTeXify */)));
+
+		// add abstract but not review
+		let abstractOrReview = doc.querySelector(".abstract");
+		if (abstractOrReview.innerText.trim().indexOf('Summary') == 0) {
+			// Strip the MathJax/MathML, put in the LaTeX math text, and
+			// surround them with $ $.
+			item.abstractNote = cleanupMath(abstractOrReview)
+				.innerText.trim().replace(/^Summary:\s*/i, "");
+		}
+
+		item.attachments = [{
+			title: "Snapshot",
+			document: doc
+		}];
+
+		let zbIDElem = doc.querySelector(".label");
+		if (zbIDElem) {
+			let zbID = cleanText(zbIDElem); // Zbl number (see also URL).
+
+			if (zbID) {
+				zbID = zbID.replace(/^Zbl\s*/i, "Zbl: ");
+				if (!item.extra) {
+					item.extra = zbID;
 				}
-				var classifications = ZU.xpath(doc, '//div[@class="classification"]//tr');
-				for (let classification of classifications) {
-					item.extra = (item.extra ? item.extra + "\n" : '') + 'MSC2010: ' + ZU.trimInternal(ZU.xpathText(classification, './td', null, " = "));
+				else {
+					item.extra += `\n${zbID}`;
 				}
-			}
-			
-			// add abstract but not review
-			var abstractOrReview = ZU.xpathText(doc, '//div[@class="abstract"]');
-			if (abstractOrReview.indexOf('Summary') == 0) {
-				item.abstractNote = abstractOrReview.replace(/^Summary:?\s*/, '');
-			}
-			
-			item.attachments = [{
-				title: "Snapshot",
-				document: doc
-			}];
 
-			var id = ZU.xpath(doc, '//div[@class="title"]/a[@class="label"]')[0];
-			if (id) {
-				if (!item.extra) item.extra = '';
-				else item.extra += "\n";
-				
-				item.extra += 'Zbl: ' + ZU.trimInternal(id.textContent)
-					.replace(/^\s*Zbl\s+/i, ''); // e.g. Zbl 1255.05045
-				item.url = id.href;
+				// zb permalink, cleaner than document URL
+				item.url = zbIDElem.href;
 			}
-			
-			item.complete();
-			// Z.debug(item);
-		});
-		
-		trans.translate();
+		}
+
+		item.complete();
 	});
+
+	trans.translate();
 }
 
+// Clean up the MathJaX-rendered text in elements. Returns a clone of the node
+// with the duplicate-causing elements removed and the LaTeX math text
+// converted to text nodes (surrounded with $ $ if laTeXify = true).
+function cleanupMath(element, laTeXify = true) {
```

Comment:
I'd want to hear from @dstillman or @AbeJellinek what Zotero's view is on handling LaTeX/MathJaX in fields. It's currently not supported, so adding things like $$ doesn't do any good, but given the nature of the translator it might still make sense?